artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/AMQPTunneledCoreMessageReader.java

```diff
@@ -155,6 +149,10 @@ public Message readBytes(Delivery delivery) {
       coreMessage.reloadPersistence(buffer, sessionSPI.getCoreMessageObjectPools());
       coreMessage.setMessageID(sessionSPI.getStorageManager().generateID());
 
-      return coreMessage;
+      serverReceiver.onMessageComplete(delivery, coreMessage, deliveryAnnotations);
+
+      close();
```

Comment:
The receiver should be closing this in onMessageComplete, a reader shouldn't close itself.

Reply:
I changed the contract there. so the thing is closing itself before calling the onComplete.

I was going to use that to release the resources.. but it's not needed any more.

Reply:
I'm saying its wrong and I'm saying its thread unsafe.  And if I need to I'm saying -1 until I'm sure it is

Reply:
..commented the same on another file before I got to this one :)

Reply:
I added a commit to be squashed addressing this